app/src/main/java/com/stevesoltys/seedvault/settings/SettingsViewModel.kt

```diff
@@ -65,7 +69,11 @@ internal class SettingsViewModel(
     }
 
     internal fun backupNow() {
-        Thread { requestBackup(app) }.start()
+        if (notificationManager.hasActiveBackupNotifications()) {
+            Toast.makeText(app, R.string.notification_backup_already_running, LENGTH_LONG).show()
```

Comment:
I'm not the biggest fan of toast messages tbh.

A snackbar might be better.

Reply:
We are in a view model here and can't show a snackbar. This is just to give some sort of feedback why we won't run a backup now and under normal circumstances, this won't be triggered (only when a user wants to manually trigger a backup, but didn't notice that one is already running). Not sure the extra effort for a snackbar would be justified.